### DISCUSSION TITLE
Slider's follow area didn't appear despite hitting the slider head

### DISCUSSION THREAD

**Discussion by soopax** [ppy/osu#23601] [Q&A] [answered]
Does anybody know why in this replay the accuracy dropped from 100.00% to 99.97% at 00:32:007 (6)? Seems like the slider's "follow area" didn't appear despite hitting the slider head, causing a miss on the slider end. I also noticed this happening when I was playing. I'm unsure if this is a bug.

May be related to https://github.com/ppy/osu/discussions/16228, but trying to import the attached replay there fails, so I can't analyze it.

The replay:
[soopax playing NayutalieN - Alien Alien (Nhato Remix) (Awlex) [Insane] (2023-05-20_19-49).zip](https://github.com/ppy/osu/files/11522817/soopax.playing.NayutalieN.-.Alien.Alien.Nhato.Remix.Awlex.Insane.2023-05-20_19-49.zip)

**Comment by monochrome22** [comment]
If I understand this correctly, the follow area didn't appear because your cursor never went on the sliderball.
You hit the slider head and moved your cursor away before the sliderball appeared.

![Notonsliderball](https://github.com/ppy/osu/assets/122969285/6109c862-7750-4848-90e9-4a4f173ddfa6)

**Reply by soopax** [reply]
In that case I should have missed the slider, right? But I got a 'great' hit.

**Reply by monochrome22** [reply]
You got a 'great' hit because you hit the slider head in the correct timing window. 

If you miss the slider head or if your cursor is not on the sliderball during a slider tick or a reverse arrow, then you will get a miss. This slider didn't have slider ticks or a reverse arrow, so you didn't get a miss, despite not having your cursor on the sliderball.

Not having your cursor on the sliderball for the slider end doesn't give you a miss or a combo break, it only gives you a missed sliderend and a slight accuracy drop.

**Reply by soopax** [reply]
Ah, I think I understand now. So the cursor was on the slider head and it was hit in the correct timing window. But I hit the slider too early and when the slider had actually started, I had already moved my cursor off of the sliderball. So I think that's why the follow area didn't appear.

I guess the question now is that if this is the case, is it intended? I feel like the follow area should still activate even if you move the cursor away from the sliderball before the slider starts. The slider head is a part of the slider after all. I don't play stable so I have no idea how this works there. This has happened before a couple times and I didn't think much of it, but when it happens it is a tiny bit unsatisfying.

tl;dr Copium skill issue (?)

**Reply by monochrome22** [reply]
> So the cursor was on the slider head and it was hit in the correct timing window. But I hit the slider too early and when the slider had actually started, I had already moved my cursor off of the sliderball. So I think that's why the follow area didn't appear.

Correct.

I don't know what the developer's sentiments are regarding the mechanic, but I can say it does work the same way in stable.

**Reply by peppy** [reply]
The specific case here is that a slider will not start "tracking" until the slider's `StartTime` is reached. If you start moving along the slider before this time, it will not give you the added lenience around the ball (ie. the follow circle size).

I'm actually not sure if this is good or not from a gameplay perspective, but if it matches stable then that's likely how we'll leave it for now.

So to clarify, it has nothing to do with ticks or anything – it's the case where you hit a slider early and don't remain on top of the ball (head circle) until it appears.

---

It would be beneficial if someone can test this on stable and confirm it matches the behaviour there. I think it does.

**Reply by monochrome22** [reply, marked as answer]
My stable test:

https://github.com/ppy/osu/assets/122969285/eb198691-bcba-4afd-a44d-ef18bce19886

**Reply by peppy** [reply]
Thanks, let's call this one resolved for now.

**Reply by soopax** [reply]
Similar weirdness happened again. Please see the attached video and replay. I remember at the time of playing, that the follow area of the slider at 01:16:551 (1) didn't appear, even after snapping the cursor back to the sliderball and starting to follow the slider properly. More importantly, the replay produced is inconsistent. When watching, sometimes the follow area is shown, but rewinding the replay back by a few seconds and watching the slider again may show it not appearing.

# Video

https://www.youtube.com/watch?v=18K-yegv644

# Replay

[soopax playing Soleily feat. SUNRaiSE - Rise Above (Nytrocide_) [-Sylvari_s Insane] (2023-06-18_21-34).osr.txt](https://github.com/ppy/osu/files/11824456/soopax.playing.Soleily.feat.SUNRaiSE.-.Rise.Above.Nytrocide_.-Sylvari_s.Insane.2023-06-18_21-34.osr.txt)

After downloading, please remove the .txt file extension from the end of the file name.

Timestamp: 01:16:551 (1)

# Versions
* osu! 2023.617.0-lazer https://github.com/ppy/osu/releases/tag/2023.617.0
* The replay also behaves inconsistently in: osu! 2023.621.0-lazer https://github.com/ppy/osu/releases/tag/2023.621.0
